toolchain/check/convert.cpp

```diff
@@ -150,14 +151,14 @@ static auto MakeElementAccessInst(Context& context, SemIR::LocId loc_id,
     // TODO: Add a new instruction kind for indexing an array at a constant
     // index so that we don't need an integer literal instruction here, and
     // remove this special case.
-    auto index_id = block.template AddInst<SemIR::IntLiteral>(
+    auto index_id = block.template AddInstReusingLoc<SemIR::IntLiteral>(
```

Comment:
Is this actually reusing a location, or is it just using `ReusingLoc` in order to bypass typechecking? If the latter, maybe we should rename `ReusingLoc` to something like `UntypedLoc`? And/or maybe we should have a TODO to plumb the typed parse node into this function so that it can use `AddInst`?

Reply:
The `loc_id` here is what's passed in. I don't know if a typed node could be plumbed, I'm guessing no due to the semantics of element access (e.g., the index could be the result of an expression). Note too the TODO above is to get rid of this, replacing with indexing at a constant index.